### PR TITLE
QR Code Scanner MVP

### PR DESCRIPTION
**Pull request by s0phialiu**
## Description

* Update the UI and flows for HackIllinois 2024 QR code scanner
* Add mentor and point shop models/scanner
* Add flow for attendee and staff
* Fix chip (staff button) UI to be more clean/cohesive with designs

## Todos
* Test QR scanning flow with team to ensure all edge cases are handled

- [x] Test attendee check in for events 
- [ ] Test mentor scanning
- [x] Test attendee scanning QR codes
- [x] Optimize iPad UI

## Screenshots
Pictured below: Attendee flow, staff flow, and guest UI

IMG_0807
IMG_0809
ub.com/HackIllinois/iOS/assets/91706701/0b435714-7bca-4cf7-ac14-2218411f4b6c)
IMG_0808

### CODE REVIEW

File: HIAPI/Services/EventService.swift [hunk 1]
```diff
@@ -51,7 +51,7 @@ public class EventService: BaseService {
     public static func getStaffCheckInEvents(authToken: String) -> APIRequest<StaffEventContainer> {
         var header = HTTPHeaders()
         header["Authorization"] = authToken
-        return APIRequest<StaffEventContainer>(service: self, endpoint: "event/filter/?displayOnStaffCheckin=true", headers: header, method: .GET)
+        return APIRequest<StaffEventContainer>(service: self, endpoint: "event/staff/", headers: header, method: .GET)
```
Comment (lasyaneti) [inline HIAPI/Services/EventService.swift]: wrong endpoint here

**Review by lasyaneti — APPROVED** [review]
